python/cudf/benchmarks/common/utils.py

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+"""Common utilities for fixture creation and benchmarking."""
+
+import inspect
+import re
+import textwrap
+from collections.abc import MutableSet
+from itertools import groupby
+from numbers import Real
+
+import pytest_cases
+from config import NUM_COLS, NUM_ROWS, cudf, cupy
+
+
+def make_gather_map(len_gather_map: Real, len_column: Real, how: str):
+    """Create a gather map based on "how" you'd like to gather from input.
+    - sequence: gather the first `len_gather_map` rows, the first thread
+                collects the first element
+    - reverse:  gather the last `len_gather_map` rows, the first thread
+                collects the last element
+    - random:   create a pseudorandom gather map
+
+    `len_gather_map`, `len_column` gets rounded to integer.
+    """
+    len_gather_map = round(len_gather_map)
+    len_column = round(len_column)
+
+    rstate = cupy.random.RandomState(seed=0)
```

Comment:
Should seed be a global configuration?

Reply:
I guess the question is how would that global seed be used? Do we use the same seed for all functions? For example does `make_boolean_mask_column` also use the same seed?

Reply:
I agree that this is mostly a concern for tests, not benchmarks, which is why I didn't think a whole lot about this yet. I think we can afford to punt on making this decision in this PR and then spend a bit of time investigating afterwards.